### PR TITLE
Implement discussion post votes

### PR DESCRIPTION
**Pull request by allenwq**
Models for post votes.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -52,16 +52,17 @@
...
     t.integer  "attachment_id",   null: false, index: {name: "fk__attachment_references_attachment_id"}, foreign_key: {references: "attachments", name: "fk_attachment_references_attachment_id", on_update: :no_action, on_delete: :no_action}
-    t.string   "name",            limit: 255, null: false
+    t.string   "name",            limit: 255,                                null: false
     t.datetime "expires_at"
     t.integer  "creator_id",      null: false, index: {name: "fk__attachment_references_creator_id"}, foreign_key: {references: "users", name: "fk_attachment_references_creator_id", on_update: :no_action, on_delete: :no_action}
     t.integer  "updater_id",      null: false, index: {name: "fk__attachment_references_updater_id"}, foreign_key: {references: "users", name: "fk_attachment_references_updater_id", on_update: :no_action, on_delete: :no_action}
     t.datetime "created_at",      null: false
     t.datetime "updated_at",      null: false
+    t.uuid     "id",              default: { expr: "uuid_generate_v4()" }, default: "uuid_generate_v4()"
```
Comment (lowjoel) [inline db/schema.rb]: I thought you didn't want to change this?

Comment (allenwq) [inline db/schema.rb]: Sorry, I think this is because switching branches...

**Comment by allenwq** [comment]
@lowjoel I got a namespace conflict, can I just put the model under discussion as `Course::Discussion::Vote` ?

**Comment by lowjoel** [comment]
What's the conflict?

**Comment by allenwq** [comment]
https://travis-ci.org/Coursemology/coursemology2/jobs/119437610
FactoryGirl complains `NameError: uninitialized constant Course::Discussion::Post::Vote`, also when I access ` Course::Discussion::Post::Vote` inside ` Course::Discussion::Post`, get the same error.

**Comment by lowjoel** [comment]
Try forward-declaring Course::Discussion::Post inside the Vote model?

**Comment by allenwq** [comment]
@lowjoel Tried a few times, get it work now !

**Comment by lowjoel** [comment]
what was the problem?

**Comment by allenwq** [comment]
@lowjoel  Thanks !

I removed those `class_name:` options and after a `rake db:reset` it works. Cannot reproduce it now, think I defined the `class_name:` wrongly, even get `undefined method creator= for Course::Discussion::Post::Vote` before.

**Pull request merged**